File: Benchmarking/src/gen_101_35.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [Double]
  }
  struct s1 {
    var p0: s0
    var p1: s0
    var p2: [[s0]]
    var p3: s0
  }
  struct s2 {
    var p0: s0
    var p1: s0
  }
  struct s6 {
    var p0: [[s0]]
    var p1: [s2]
  }
  func f27(_ v0: Double) -> Double {
    var v3: Double = v0
    var v5: Double = v0
    var v2: Double = v3
    v5 = v3
    let v13: Double = v0 + v2
    var v6: Double = v3
    var v8: Double = v13
    let v7: Double = v2 * v0
    var v1: Double = v6
    var v10: Double = v3
    var v11: Double = v0
    v2 = v13
    var v4: Double = v5
    var v14: Double = v4
    let v9: [Double] = [v7, v7, v3, v10, v11]
    var v12: [Double] = v9
    var v17: [Double] = v12
    var v23: Double = v1
    v17[3] = v6
    v17[0] = v23
    var v42: Double = v0
    v10 = v42
    v17[2] = v8
    let v27: Double = v17[1]
    v17[3] = v14
    return v27
  }
  @inline(never)
  func f0(_ v0: s6, _ v1: [s1], _ v2: Double) -> Double {
    var v8: Double = v2
    let v30: Double = v8 / v8
    let v41: Double = f27(v2)
    let v68: Double = v30 + v41
    var v182: Double = v41
    let v175: Double = v68 * v182
    return v175
  }
  func benchmark() {
    let v0: s6 = s6(p0: [[s0(p0: [[0.0]], p1: [1.0])]], p1: [s2(p0: s0(p0: [[2.0]], p1: [3.0]), p1: s0(p0: [[4.0]], p1: [5.0]))])
    let v1: [s1] = [s1(p0: s0(p0: [[6.0]], p1: [7.0]), p1: s0(p0: [[8.0]], p1: [9.0]), p2: [[s0(p0: [[10.0]], p1: [11.0])], [s0(p0: [[12.0]], p1: [13.0])], [s0(p0: [[14.0]], p1: [15.0])]], p3: s0(p0: [[16.0]], p1: [17.0])), s1(p0: s0(p0: [[18.0]], p1: [19.0]), p1: s0(p0: [[20.0]], p1: [21.0]), p2: [[s0(p0: [[22.0]], p1: [23.0])], [s0(p0: [[24.0]], p1: [25.0])], [s0(p0: [[26.0]], p1: [27.0])]], p3: s0(p0: [[28.0]], p1: [29.0])), s1(p0: s0(p0: [[30.0]], p1: [31.0]), p1: s0(p0: [[32.0]], p1: [33.0]), p2: [[s0(p0: [[34.0]], p1: [35.0])], [s0(p0: [[36.0]], p1: [37.0])], [s0(p0: [[38.0]], p1: [39.0])]], p3: s0(p0: [[40.0]], p1: [41.0]))]
    let v2: Double = 42.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 42.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()

File: Benchmarking/src/gen_101_35.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<double> p1;
    s0(std::vector<std::vector<double>> v0, std::vector<double> v1): p0(v0), p1(v1) { }
  };
  struct s1 {
    s0 p0;
    s0 p1;
    std::vector<std::vector<s0>> p2;
    s0 p3;
    s1(s0 v0, s0 v1, std::vector<std::vector<s0>> v2, s0 v3): p0(v0), p1(v1), p2(v2), p3(v3) { }
  };
  struct s2 {
    s0 p0;
    s0 p1;
    s2(s0 v0, s0 v1): p0(v0), p1(v1) { }
  };
  struct s6 {
    std::vector<std::vector<s0>> p0;
    std::vector<s2> p1;
    s6(std::vector<std::vector<s0>> v0, std::vector<s2> v1): p0(v0), p1(v1) { }
  };
  double f27(const double &v0) {
    double v3 = v0;
    double v5 = v0;
    double v2 = v3;
    v5 = v3;
    const double v13 = v0 + v2;
    double v6 = v3;
    double v8 = v13;
    const double v7 = v2 * v0;
    double v1 = v6;
    double v10 = v3;
    double v11 = v0;
    v2 = v13;
    double v4 = v5;
    double v14 = v4;
    const std::vector<double> v9 { v7, v7, v3, v10, v11 };
    std::vector<double> v12 = v9;
    std::vector<double> v17 = v12;
    double v23 = v1;
    v17[3] = v6;
    v17[0] = v23;
    double v42 = v0;
    v10 = v42;
    v17[2] = v8;
    const double v27 = v17[1];
    v17[3] = v14;
    return v27;
  }
  __attribute__((noinline))
  double f0(const s6 &v0, const std::vector<s1> &v1, const double &v2) {
    double v8 = v2;
    const double v30 = v8 / v8;
    const double v41 = f27(v2);
    const double v68 = v30 + v41;
    double v182 = v41;
    const double v175 = v68 * v182;
    return v175;
  }
  int main() {
    s6 v0({ { { { { { 0.0 } }, { 1.0 } } } }, { { { { { 2.0 } }, { 3.0 } }, { { { 4.0 } }, { 5.0 } } } } });
    std::vector<s1> v1({ { { { { 6.0 } }, { 7.0 } }, { { { 8.0 } }, { 9.0 } }, { { { { { 10.0 } }, { 11.0 } } }, { { { { 12.0 } }, { 13.0 } } }, { { { { 14.0 } }, { 15.0 } } } }, { { { 16.0 } }, { 17.0 } } }, { { { { 18.0 } }, { 19.0 } }, { { { 20.0 } }, { 21.0 } }, { { { { { 22.0 } }, { 23.0 } } }, { { { { 24.0 } }, { 25.0 } } }, { { { { 26.0 } }, { 27.0 } } } }, { { { 28.0 } }, { 29.0 } } }, { { { { 30.0 } }, { 31.0 } }, { { { 32.0 } }, { 33.0 } }, { { { { { 34.0 } }, { 35.0 } } }, { { { { 36.0 } }, { 37.0 } } }, { { { { 38.0 } }, { 39.0 } } } }, { { { 40.0 } }, { 41.0 } } } });
    double v2(42.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1, v2);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }
